packages/test/test-end-to-end-tests/src/test/sharedStringEndToEndTests.spec.ts

```diff
@@ -68,4 +71,36 @@ describeFullCompat("SharedString", (getTestObjectProvider) => {
         assert.equal(
             newSharedString.getText(), text, "The new container should receive the inserted text on creation");
     });
+
+    it("marker passes on attachment directly and transitively to any referenced DDS", async () => {
+        sharedString1.insertText(0, "hello world");
+        // Insert a simple marker.
+        sharedString1.insertMarker(
+            0,
+            ReferenceType.Simple,
+            {
+                [reservedMarkerIdKey]: "markerId",
+            },
+        );
+
+        // When an unattached DDS refers to another unattached DDS, both remain unattached
+        const detachedString = SharedString.create(dataObject1.runtime, "detachedString");
+        detachedString.insertText(0, "blue");
+        const detachedMap: ISharedMap = SharedMap.create(dataObject1.runtime);
```

Comment:
Why have the intermediary DDS be a SharedMap instead of a SharedString to exercise sequence attachment code since map is already covered elsewhere?

Reply:
To test transitivity. A SharedString can't have a DDS handle inside it can it? Or is the idea to do:

SharedString[Marker] -> SharedString[Marker].handle -> SharedString.handle?

Reply:
Yes, that.